subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -1195,20 +1245,53 @@ enum hawkbit_response hawkbit_probe(void)
 
 	flash_img_init(&hb_context.flash_ctx);
 
-	if (!send_request(HTTP_GET, HAWKBIT_DOWNLOAD,
+	hash_info = mbedtls_md_info_from_type(MBEDTLS_MD_SHA256);
+	if (!hash_info) {
+		LOG_ERR("Unable to request hash type from mbedTLS");
+		hb_context.code_status = HAWKBIT_METADATA_ERROR;
+		goto cleanup;
+	}
+
+	mbedtls_md_init(&hb_context.dl.hash_ctx);
+	if (mbedtls_md_setup(&hb_context.dl.hash_ctx, hash_info, 0) < 0) {
+		LOG_ERR("Can't setup mbedTLS hash engine");
+		mbedtls_md_free(&hb_context.dl.hash_ctx);
+		hb_context.code_status = HAWKBIT_METADATA_ERROR;
+		goto free_md;
+	}
+
+	mbedtls_md_starts(&hb_context.dl.hash_ctx);
+
+	ret = (int)send_request(HTTP_GET, HAWKBIT_DOWNLOAD,
 			  HAWKBIT_STATUS_FINISHED_NONE,
-			  HAWKBIT_STATUS_EXEC_NONE)) {
+			  HAWKBIT_STATUS_EXEC_NONE);
+
+	mbedtls_md_finish(&hb_context.dl.hash_ctx,
+			  response_hash);
+
+	if (!ret) {
 		LOG_ERR("Send request failed");
 		hb_context.code_status = HAWKBIT_NETWORKING_ERROR;
-		goto cleanup;
+		goto free_md;
 	}
 
 	if (hb_context.code_status == HAWKBIT_DOWNLOAD_ERROR) {
-		goto cleanup;
+		goto free_md;
 	}
 
-	if (boot_request_upgrade(BOOT_UPGRADE_TEST)) {
-		LOG_ERR("Download failed");
+	if (!hb_context.final_data_received) {
+		LOG_ERR("Download is not complete");
+		hb_context.code_status = HAWKBIT_DOWNLOAD_ERROR;
+	} else if (memcmp(response_hash, hb_context.dl.file_hash,
```

Comment:
Does hawkbit server supply file hash info on its json response with available docker image?

Reply:
yes the available docker image provides file hash info in three format's `sha1`, `md5` and `sha256`.

Reply:
@sylvioalves It [should](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/mgmt/hawkbit/hawkbit.c#L588). You can verify if you [fixed and enabled the log](https://github.com/zephyrproject-rtos/zephyr/pull/37351/commits/d761a3a74816cf543fb16c671d9f8c532e69139a).

Reply:
Available docker image provides file hash info in three format's `sha1`, `md5` and `sha256`